### PR TITLE
Proposed a new article about using pure PHP libraries with Assetic

### PR DESCRIPTION
**Pull request by javiereguiluz**
| Q | A |
| --- | --- |
| Doc fix? | no |
| New docs? | yes |
| Applies to | all |
| Fixed tickets | - |

In #5159 @WouterJ proposes to create a new Cookbook section dedicated to frontend articles. I think it's a great idea and in this pull request I propose an article showing how to use pure PHP libraries to combine, compile and minimize CSS, SCSS and JS files.

### CODE REVIEW

File: cookbook/frontend/assetic_php.rst [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+.. code-block:: html+jinja
+
+    <!DOCTYPE html>
+    <html>
+        <head>
+            <!-- ... -->
+
+            {% stylesheets filter="scssphp" output="css/app.css"
+                "assets/scss/bootstrap.scss"
+                "assets/scss/font-awesome.scss"
```
Comment (Rvanlaak) [inline cookbook/frontend/assetic_php.rst]: Maybe add a reference about managing FontAwesome using Bower here?

Comment (javiereguiluz) [inline cookbook/frontend/assetic_php.rst]: I don't think that's a good idea, because this article is about using PHP only libraries. But once the Bower article written by @WouterJ is published, we can indeed add a link to it in the introduction of this article (where we say "JS is the best  for this"). I think taht would make a lot of sense. Thanks!

Comment (Rvanlaak) [inline cookbook/frontend/assetic_php.rst]: :+1: for adding a note. My point is that it should be clear that managing (minimize, compile) web assets with PHP is possible, but checking in "assets/scss/font-awesome.scss" in the VCS is not needed if a package manager like Bower is used.

File: cookbook/frontend/assetic_php.rst [hunk 2]
```diff
@@ -0,0 +1,152 @@
+.. index::
+    single: Front-end; Assetic, Bootstrap
+
+Combining, Compiling and Minimizing Web Assets with Pure PHP Libraries
```
Comment (cordoval) [inline cookbook/frontend/assetic_php.rst]: should say PHP Libraries only

File: cookbook/frontend/assetic_php.rst [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+technical point of view, using pure PHP alternative libraries can be useful in
+some scenarios:
+
+* If you can't install or use ``npm`` and the other JavaScript solutions;
+* If you prefer to limit the amount of different technologies used in your
+  applications;
+* If you want to simplify application deployment.
+
+In this article you'll learn how to combine and minimize CSS and JavaScript files
+and how to compile Sass SCSS files using pure PHP libraries with Assetic.
```
Comment (cordoval) [inline cookbook/frontend/assetic_php.rst]: pure -> only

File: cookbook/frontend/assetic_php.rst [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+
+Assetic includes a lot of ready-to-use filters but it doesn't include their
+associated libraries. Therefore, before enabling the filters used in this article
+you must install two libraries. Open a command console, browse to your project
+directory and execute the following commands:
+
+.. code-block:: bash
+
+    $ composer require leafo/scssphp
+    $ composer require patchwork/jsqueeze:"~1.0"
```
Comment (cordoval) [inline cookbook/frontend/assetic_php.rst]: make it ~1.0.7 better?

File: cookbook/frontend/assetic_php.rst [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+
+It's very important to maintain the ``~1.0`` version constraint for the ``jsqueeze``
+dependency because the most recent stable version is not compatible with Assetic.
+
+Organizing Your Web Asset Files
+-------------------------------
+
+This example shows the very common scenario of using the Bootstrap framework, the
+jQuery library, the FontAwesome icon font and some regular CSS and JavaScript
```
Comment (cordoval) [inline cookbook/frontend/assetic_php.rst]: font -> fonts?

File: cookbook/frontend/assetic_php.rst (hunk 1 above)
Comment (cordoval) [inline cookbook/frontend/assetic_php.rst]: :+1: interesting point, yeah maybe a # this is required when done PHP exclusive compilation of assets

File: cookbook/frontend/assetic_php.rst (hunk 5 above)
Comment (javiereguiluz) [inline cookbook/frontend/assetic_php.rst]: In this case we're talking about just one font, so I guess this should be kept as singular.

File: cookbook/frontend/assetic_php.rst (hunk 4 above)
Comment (javiereguiluz) [inline cookbook/frontend/assetic_php.rst]: When we want to limit something to be *"less than 2.0.0"* I think that the `~1.0` constraint is more intuitive.

File: cookbook/frontend/assetic_php.rst (hunk 5 above)
Comment (cordoval) [inline cookbook/frontend/assetic_php.rst]: i guess you are right https://github.com/FortAwesome/Font-Awesome/tree/master/fonts

File: cookbook/frontend/assetic_php.rst (hunk 4 above)
Comment (cordoval) [inline cookbook/frontend/assetic_php.rst]: actually is more correct, not intuitive, because it will include 1.2 etc. Though is very unlikely. So ignore please and proceed :+1:

**Comment by Rvanlaak** [comment]
As mentioned in https://github.com/symfony/symfony-docs/pull/5166#discussion_r28228404 What about compiling a file that is managed by a package manager?

The example I'd like to be clear is: [the scss files from FontAwesome](https://github.com/FortAwesome/Font-Awesome/tree/master/scss) can be managed using `Bower`, and can be compiled using `scssphp`.

File: cookbook/frontend/assetic_php.rst (hunk 5 above)
Comment (javiereguiluz) [inline cookbook/frontend/assetic_php.rst]: Actually ... I think you are right. FontAwesome is just one font, but it delivers a lot of files. Strictly speaking, there are several icon fonts. I've just updated it. Thanks.

**Comment by javiereguiluz** [comment]
@Rvanlaak let me explain the purpose of this article: the idea is to explain how to do the usual web asset stuff WITHOUT using any JavaScript tool or technology (Bower, Gulp, Grunt).

This article shows how can you get everything with Assetic and two PHP libraries while setting up everything in just a few miutes. So the FontAwesome files, as long as Bootstrap and jQuery files, should be committed to the repository and stored in the `web/assets/` directory, as explained in the article.

**Comment by cordoval** [comment]
i think this article should include packages such as robloach component or other bundles that bring in third party stuff. The path should be adaptable to point to those things without needing to persist the assets into the repository which is a very bad practice.

**Comment by javiereguiluz** [comment]
@cordoval I'm afraid that adding JavaScript stuff or third-party bundles into this article can't be possible.

The intro of the article says: *"using JavaScript for doing this stuff is the best way to do it"* and provides a link to that solution. But if you can't or don't want to use JavaScript, then you have this article which explains the simple and pragmatic way to do it using only PHP.

**Comment by cordoval** [comment]
I think you are not understanding what i am saying. Let me rephrase it and reiterate it to make it plain.

I mentioned @robloach's component only in the sense to bring css in, it has php packages that bring assets and js as well. But they are not the js method to bring assets in. These are purely php. That said I hope is clear up to this point. I was never talking about js way of bringing assets here, which I am afraid you misunderstood.

That said, there is no need then to persist these things into the repository, or at least there is no need to persist them all. It can be done right with php as well, or the most right way with only php.

**Comment by javiereguiluz** [comment]
@cordoval I understand you know. Thanks for explaining it in detail.

However, I still think it's not necessary. In this case, I don't consider a bad practice to persist assets in the repository. If your application is complex, use Bower and JS-stuff to continuosly update those assets. But if you application is simple enough, it's better to not update them continuously: things can break and you don't have time/money to solve those problems. So it's better to add those assets to the repo and treat them like regular assets, without updating them continuously.

**Comment by cordoval** [comment]
I meant to put just a note, not even a full paragraph, to avoid saying this is the only way as there are the ways i just explained.

**Comment by javiereguiluz** [comment]
@cordoval I agree. We'll do that.

**Comment by Rvanlaak** [comment]
:+1: for not using any JS tooling in the article, but following a PHP approach. Did get that, but did want to discuss the gray areas where developers could make a decision to use a slight other approach. The use of bower is only one example of that ;) I don't see any other areas atm, so adding a note/link to @WouterJ 's Bower article is fine ;)

**Comment by cordoval** [comment]
@Rvanlaak i think @javiereguiluz understood what i meant. PHP approach is not just one approach is manyfold. I think this article is saying this is the best way, but if you are going to do it plainly bare bones here are the ways people approach it.

File: cookbook/frontend/index.rst [hunk 6]
```diff
@@ -0,0 +1,7 @@
+Form
```
Comment (xabbuh) [inline cookbook/frontend/index.rst]: should be *Frontend*

File: cookbook/frontend/assetic_php.rst [hunk 7]
```diff
@@ -0,0 +1,152 @@
+.. index::
+    single: Front-end; Assetic, Bootstrap
+
+Combining, Compiling and Minimizing Web Assets with PHP Libraries
+=================================================================
+
+The official Symfony Best Practices recommend to use Assetic to
+:doc:`manage web assets </best_practices/web-assets>`, unless you are
+comfortable with JavaScript-based frontend tools.
```
Comment (xabbuh) [inline cookbook/frontend/assetic_php.rst]: front-end

File: cookbook/frontend/assetic_php.rst [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+Even if those JavaScript-based solutions are the most suitable ones from a
+technical point of view, using pure PHP alternative libraries can be useful in
+some scenarios:
+
+* If you can't install or use ``npm`` and the other JavaScript solutions;
+* If you prefer to limit the amount of different technologies used in your
+  applications;
+* If you want to simplify application deployment.
+
+In this article you'll learn how to combine and minimize CSS and JavaScript files
```
Comment (xabbuh) [inline cookbook/frontend/assetic_php.rst]: In this article, you'll learn [...]

File: cookbook/frontend/assetic_php.rst [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+and how to compile Sass SCSS files using PHP only libraries with Assetic.
```
Comment (xabbuh) [inline cookbook/frontend/assetic_php.rst]: I'm no front-end expert, but is Sass SCSS the state-of-the-art technology? What about alternatives like LESS?

File: cookbook/frontend/assetic_php.rst [hunk 10]
```diff
@@ -0,0 +1,152 @@
...
+
+Installing the Third-Party Compression Libraries
+------------------------------------------------
+
+Assetic includes a lot of ready-to-use filters but it doesn't include their
```
Comment (xabbuh) [inline cookbook/frontend/assetic_php.rst]: [...] filters, but it [...]

File: cookbook/frontend/assetic_php.rst [hunk 11]
```diff
@@ -0,0 +1,152 @@
...
+associated libraries. Therefore, before enabling the filters used in this article
```
Comment (xabbuh) [inline cookbook/frontend/assetic_php.rst]: [...] in this article, you must [...]

File: cookbook/frontend/assetic_php.rst [hunk 12]
```diff
@@ -0,0 +1,152 @@
...
+
+.. code-block:: bash
+
+    $ composer require leafo/scssphp
+    $ composer require patchwork/jsqueeze:"~1.0"
+
+It's very important to maintain the ``~1.0`` version constraint for the ``jsqueeze``
+dependency because the most recent stable version is not compatible with Assetic.
+
+Organizing Your Web Asset Files
```
Comment (xabbuh) [inline cookbook/frontend/assetic_php.rst]: Actually, @WouterJ correct me that *your* (lowercased) is the correct spelling (it's a closed-class word).

File: cookbook/frontend/assetic_php.rst [hunk 13]
```diff
@@ -0,0 +1,152 @@
...
+        │   └── mixins
+        │       ├── _alerts.scss
+        │       ├── ...
+        │       └── _vendor-prefixes.scss
+        ├── bootstrap.scss
+        ├── font-awesome
+        │   ├── _animated.scss
+        │   ├── ...
+        │   └── _variables.scss
+        └── font-awesome.scss
```
Comment (xabbuh) [inline cookbook/frontend/assetic_php.rst]: What is the reasoning behing storind the raw files (`*.scss` and not-minified JavaScript files) under `web/assets`?

File: cookbook/frontend/assetic_php.rst [hunk 14]
```diff
@@ -0,0 +1,152 @@
...
+formatter used by the filter to produce the compiled CSS file. Using the
+compressed formatter allows to minimize the resulting file, no matter if the
+original files are regular CSS files or SCSS files.
+
+Then, update the code of your Twig template to add the ``{% stylesheets %}`` tag
+defined by Assetic:
+
+.. code-block:: html+jinja
+
+    <!DOCTYPE html>
```
Comment (xabbuh) [inline cookbook/frontend/assetic_php.rst]: It's probably a good idea to add a filename comment to the top of the template.

**Comment by xabbuh** [comment]
This article overlaps in a way with the articles from the *Assetic* cookbook section. We should find a way not to document thing more than once.

File: cookbook/frontend/assetic_php.rst (hunk 8 above)
Comment (cordoval) [inline cookbook/frontend/assetic_php.rst]: I honestly don't think this way, but I am eager to learn, what is the reasoning for this comma?

Comment (WouterJ) [inline cookbook/frontend/assetic_php.rst]: It's to allow this kind of crazy word order in the sentence. Without comma, you would do "You'll learn how to combine and minimize CSS and JS files in this article" (time and location are always located at the end of the sentence). This is perfectly fine when you're speaking to someone, but when reading, it's common to put the location in front of the sentence. When doing that, you must add a comma before you start the sentence: "In this article, you'll learn [...]"

File: cookbook/frontend/assetic_php.rst (hunk 9 above)
Comment (WouterJ) [inline cookbook/frontend/assetic_php.rst]: There are some alternatives, but I believe Sass is used the most and has the biggest community behind it.

Comment (WouterJ) [inline cookbook/frontend/assetic_php.rst]: However, I would say "Sass files". There is no need to make this overcomplicated (and using uppercase words breaks up a sentence, reducing the reading experience a bit).

File: cookbook/frontend/assetic_php.rst (hunk 8 above)
Comment (cordoval) [inline cookbook/frontend/assetic_php.rst]: can you show me which of these cases is http://grammar.ccc.commnet.edu/grammar/commas.htm ? I am trying to contest this appreciation constructively :older_man:

File: cookbook/frontend/assetic_php.rst [hunk 15]
```diff
@@ -0,0 +1,152 @@
...
+In this article you'll learn how to combine and minimize CSS and JavaScript files
+and how to compile Sass SCSS files using PHP only libraries with Assetic.
+
+Installing the Third-Party Compression Libraries
+------------------------------------------------
+
+Assetic includes a lot of ready-to-use filters but it doesn't include their
+associated libraries. Therefore, before enabling the filters used in this article
+you must install two libraries. Open a command console, browse to your project
+directory and execute the following commands:
```
Comment (WouterJ) [inline cookbook/frontend/assetic_php.rst]: I think it's very important to not make this article just a step-by-step solution for one problem (Sass and minifying), but make it ready to use for other libraries as well.

This mostly means that we should try to not say "Do this", but instead explain what "this" is and how we've come up with "this". In this case, it means explaining how one can find these package names. How does one know it has to install exactly these packages?

File: cookbook/frontend/assetic_php.rst [hunk 16]
```diff
@@ -0,0 +1,152 @@
...
+
+.. code-block:: bash
+
+    $ composer require leafo/scssphp
+    $ composer require patchwork/jsqueeze:"~1.0"
+
+It's very important to maintain the ``~1.0`` version constraint for the ``jsqueeze``
+dependency because the most recent stable version is not compatible with Assetic.
```
Comment (WouterJ) [inline cookbook/frontend/assetic_php.rst]: This should be handled by Assetic, introducing a `conflict` rule in their `composer.json`.

File: cookbook/frontend/assetic_php.rst [hunk 17]
```diff
@@ -0,0 +1,152 @@
...
+
+Organizing Your Web Asset Files
+-------------------------------
+
+This example shows the very common scenario of using the Bootstrap framework, the
```
Comment (WouterJ) [inline cookbook/frontend/assetic_php.rst]: -1 for adding "very common". It's starting to look like Symfony is sponsered by Bootstrap :)

File: cookbook/frontend/assetic_php.rst (hunk 13 above)
Comment (WouterJ) [inline cookbook/frontend/assetic_php.rst]: big -1 for that. It's also against the best practices btw. Imo, uncompiled files should live in `app/Resources/public` and only the compiled onces in `web/`.

File: cookbook/frontend/assetic_php.rst [hunk 18]
```diff
@@ -0,0 +1,152 @@
...
+                "assets/scss/bootstrap.scss"
+                "assets/scss/font-awesome.scss"
+                "assets/css/*.css"
+            %}
+                <link rel="stylesheet" href="{{ asset_url }}" />
+            {% endstylesheets %}
+
+This simple configuration compiles the SCSS files into regular CSS files, combines
+all of them, minimizes the contents and saves the output in the ``web/css/app.css``
+file, which is the one that is served to your visitors.
```
Comment (WouterJ) [inline cookbook/frontend/assetic_php.rst]: Huston, we've a big sentence again :)

What do you think about: "This simple configuration compiles, combines and minifies the SCSS files into a regular CSS file that's put in ``web/css/app.css``."

File: cookbook/frontend/assetic_php.rst (hunk 13 above)
Comment (cordoval) [inline cookbook/frontend/assetic_php.rst]: I don't think so, well maybe best practice document says so for newcomers. Also not following best practices does not mean wrong as it seems you are conveying with the -1. Mind that since we are opening the doors at different possibilities from where the assets come from we cannot limit them, be it bundles yes, be it other packages who knows where they are, puli, or other cases will depend on which php package is used to pull the assets no?

File: cookbook/frontend/assetic_php.rst [hunk 19]
```diff
@@ -0,0 +1,152 @@
...
+
+First, configure a new ``jsqueeze`` Assetic filter as follows:
+
+.. code-block:: yaml
+
+    # app/config/config.yml
+    assetic:
+        filters:
+            jsqueeze: ~
+            # ...
```
Comment (WouterJ) [inline cookbook/frontend/assetic_php.rst]: we should show other formats:

```
<!-- app/config/config.xml -->
<?xml version="1.0" charset="UTF-8" ?>
<container xmlns="http://symfony.com/schema/dic/services"
    xmlns:assetic="http://symfony.com/schema/dic/assetic">

    <assetic:config>
        <filter name="jsqueeze" />
        <!-- ... -->
    </assetic:config>
</container>
```

```
// app/config/config.php
$container->loadFromExtension('assetic', array(
    'filters' => array(
         'jsqueeze' => null,
         // ...
    ),
));
```

File: cookbook/frontend/assetic_php.rst [hunk 20]
```diff
@@ -0,0 +1,152 @@
...
+        {% javascripts filter="?jsqueeze" output="js/app.js"
+            "assets/js/jquery.js"
+            "assets/js/bootstrap.js"
+            "assets/js/main.js"
+        %}
+            <script src="{{ asset_url }}"></script>
+        {% endjavascripts %}
+
+        </body>
+    </html>
```
Comment (WouterJ) [inline cookbook/frontend/assetic_php.rst]: we should show the PHP syntax too

File: cookbook/frontend/index.rst (hunk 6 above)
Comment (WouterJ) [inline cookbook/frontend/index.rst]: this will be done by my PR and can be removed from this PR, to avoid merge conflicts

File: cookbook/frontend/assetic_php.rst [hunk 21]
```diff
@@ -0,0 +1,152 @@
...
+First, configure a new ``scssphp`` Assetic filter as follows:
+
+.. code-block:: yaml
+
+    # app/config/config.yml
+    assetic:
+        filters:
+            scssphp:
+                formatter: "Leafo\\ScssPhp\\Formatter\\Compressed"
+            # ...
```
Comment (WouterJ) [inline cookbook/frontend/assetic_php.rst]: we have to show XML and PHP too:

```
<!-- app/config/config.xml -->
<?xml version="1.0" charset="UTF-8" ?>
<container xmlns="http://symfony.com/schema/dic/services"
    xmlns:assetic="http://symfony.com/schema/dic/assetic">

    <assetic:config>
        <filter name="scssphp" formatter="Leafo\ScssPhp\Formatter\Compressed" />
        <!-- ... -->
    </assetic:config>
</container>
```

```
// app/config/config.php
$container->loadFromExtension('assetic', array(
    'filters' => array(
         'jsqueeze' => null,
         // ...
    ),
));
```

File: cookbook/frontend/assetic_php.rst [hunk 22]
```diff
@@ -0,0 +1,152 @@
...
+
+First, configure a new ``scssphp`` Assetic filter as follows:
+
+.. code-block:: yaml
+
+    # app/config/config.yml
+    assetic:
+        filters:
+            scssphp:
+                formatter: "Leafo\\ScssPhp\\Formatter\\Compressed"
```
Comment (WouterJ) [inline cookbook/frontend/assetic_php.rst]: escaping the `\` isn't needed and as it reduces readability, we always try to avoid it in the docs.

**Comment by WouterJ** [comment]
Imo, as this is about Assetic, we should move this one to `cookbook/assetic`. I've created `cookbook/frontend` exactly to have front-end docs that are *not* about Assetic (as we only had Assetic docs until now).

File: cookbook/frontend/assetic_php.rst (hunk 13 above)
Comment (WouterJ) [inline cookbook/frontend/assetic_php.rst]: My -1 was for putting them in the public files (`web/`). My personal opinion is to put them in `app/Resources/public`.

File: cookbook/frontend/assetic_php.rst (hunk 8 above)
Comment (WouterJ) [inline cookbook/frontend/assetic_php.rst]: It's 3:

> 
Use a comma to set off introductory elements, as in "Running toward third base, he suddenly realized how stupid he looked."

> It is permissible to omit the comma after a brief introductory element if the omission does not result in confusion or hesitancy in reading. If there is ever any doubt, use the comma, as it is always correct. If you would like some additional guidelines on using a comma after introductory elements, click HERE.

So, yes. You are correct that the comma can be omitted, but it can introduce confusion. As the docs are read by many non-natives, it's good to avoid anything that could introduce confusion.

Comment (cordoval) [inline cookbook/frontend/assetic_php.rst]: well you agreed which is good. So it is not "it must" but it is optional. If we make this into a rule just to be always right is ok. I personally think this interlude is so short that flows immediately to the point, no need for refrain or pause. I also doubt this introduces any confusion. But :green_heart: :traffic_light:  light if you always want to be right. It is just i am concern on the damage to the language over time with enforcing something that is optional indeed. But yeah let's go in agreement to put the comma :older_man:

File: cookbook/frontend/assetic_php.rst (hunk 4 above)
Comment (stof) [inline cookbook/frontend/assetic_php.rst]: I was about to suggest using the 2.0 version, but actually Assetic is compatible with both 1.x and 2.x only in the dev version. I haven't released this in a stable version yet.

File: cookbook/frontend/assetic_php.rst (hunk 16 above)
Comment (stof) [inline cookbook/frontend/assetic_php.rst]: The dev version of Assetic is compatible with both 1.x and 2.x actually

File: cookbook/frontend/assetic_php.rst (hunk 22 above)
Comment (stof) [inline cookbook/frontend/assetic_php.rst]: if you use the double-quoted syntax, it is.

Comment (WouterJ) [inline cookbook/frontend/assetic_php.rst]: actually, you need the single quoted syntax, don't you? As double quotes can cause problems when `\S`, `\F` or `\C` are valid sequences? (there are a couple of uppercase sequence names)

Comment (stof) [inline cookbook/frontend/assetic_php.rst]: @WouterJ it is fine here as the backslash is escaped. The parser will read ``\\`` (as ``\``) and then ``F``, not ``\F``

Comment (xabbuh) [inline cookbook/frontend/assetic_php.rst]: I would use single quotes so that we don't need to escape backslashes.

File: cookbook/frontend/assetic_php.rst (hunk 8 above)
Comment (javiereguiluz) [inline cookbook/frontend/assetic_php.rst]: Thanks for the discussion. I'll put the comma then :)

File: cookbook/frontend/assetic_php.rst (hunk 13 above)
Comment (javiereguiluz) [inline cookbook/frontend/assetic_php.rst]: I'm a bit lost with your latest suggestion:

  * In https://github.com/symfony/symfony-demo/issues/5 you asked my why I put the assets in `app/Resources/` and I moved them to `web/` here: https://github.com/symfony/symfony-demo/pull/33
  * Here I've put the assets in `web/` and you ask me to put them in `app/Resources/`

File: cookbook/frontend/assetic_php.rst (hunk 14 above)
Comment (javiereguiluz) [inline cookbook/frontend/assetic_php.rst]: I've updated it to display `{# app/Resources/views/base.html.twig #}`  Is that the common practice in the Symfony docs when referring to the base/layout template?

File: cookbook/frontend/assetic_php.rst (hunk 13 above)
Comment (Rvanlaak) [inline cookbook/frontend/assetic_php.rst]: Files that should be compiled using Assetic should be in `/app/Resources/` right?

File: cookbook/frontend/assetic_php.rst (hunk 17 above)
Comment (javiereguiluz) [inline cookbook/frontend/assetic_php.rst]: I've replaced `very common` by `common`.

File: cookbook/frontend/assetic_php.rst (hunk 18 above)
Comment (javiereguiluz) [inline cookbook/frontend/assetic_php.rst]: Your version is much better. Thanks!

File: cookbook/frontend/assetic_php.rst (hunk 19 above)
Comment (javiereguiluz) [inline cookbook/frontend/assetic_php.rst]: Thanks for providing these alternative formats.

**Comment by javiereguiluz** [comment]
@WouterJ I think you are right and this article should be moved back to the Assetic cookbook section instead of adding it to the new Frontend section. I've just changed it. Thanks.

**Comment by javiereguiluz** [comment]
This PR seems to be finished. I've addressed every comment and implemented every suggestion. Can we label it as "finished"? Thanks!

**Comment by xabbuh** [comment]
looks good :+1:

**Comment by weaverryan** [comment]
Merged! Nice job making this short and concise - that's the best part (and a good reason why someone might opt for the pure PHP solutions). I made some final minor tweaks at sha: bf18f16c82e74c8b8169adfe67b1b7634226bf1b

Thanks!
